utilities/serving_runtime.py

```diff
@@ -182,9 +193,22 @@ def update_model_dict(self) -> dict[str, Any]:
                     elif is_raw:
                         container["ports"] = vLLM_CONFIG["port_configurations"]["raw"]
 
+            if self.containers and container_name in self.containers:
+                container_config = self.containers.get(container_name, {})
+                for key, value in container_config.items():
+                    container[key] = value
+
+        if self.containers:
+            for container_name, container_config in self.containers.items():
```

Comment:
i think that if you deep copy `self.containers` and use `pop` when iterating over `for container in template_containers:` then here you will have a reduced dict just to add additional containers and the code will be simpler and then you can drop
`if container_name not in {container.get("name") for container in template_containers}:`

wdyt?